### PR TITLE
ISPN-11280 More ConditionFuture updates

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.redhat.com/browse/ISPN-11280

Apply #7853 feedback

### CODE REVIEW

File: core/src/main/java/org/infinispan/util/concurrent/ConditionFuture.java [hunk 1]
```diff
@@ -100,17 +100,19 @@ public void updateAsync(T value, Executor executor) {
...
-            future.completeExceptionally(t);
-         }
+         completeAllExceptionally(t);
+      }
+   }
+
+   private void completeAllExceptionally(Throwable t) {
+      List<Data> completed;
+      synchronized (futures) {
+         completed = new ArrayList<>(futures.size());
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/util/concurrent/ConditionFuture.java]: why not use `new ArrayList<>(futures.values());`

File: core/src/test/java/org/infinispan/test/Exceptions.java [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package org.infinispan.test;
 
 import java.util.concurrent.Callable;
+import java.util.concurrent.CompletionException;
```
Comment (pruivo) [inline core/src/test/java/org/infinispan/test/Exceptions.java]: unused import.

File: core/src/main/java/org/infinispan/util/concurrent/ConditionFuture.java (hunk 1 above)
Comment (danberindei) [reply core/src/main/java/org/infinispan/util/concurrent/ConditionFuture.java]: Done

**Comment by pruivo** [comment]
@danberindei checkstyle issues in CI.

**Comment by danberindei** [comment]
I finally fixed it @pruivo

**Comment by pruivo** [comment]
integrated! thanks @danberindei !

**Pull request merged**